DEVELOPMENT.md

```diff
@@ -1,109 +1,87 @@
-
-![github.com/settings/tokens with the extra space character highlighted][i:The gosh-darn whitespace]
-
-If you run into trouble, you can always fall back to typing the personal access code by hand.
-
-## Installing the Angular command-line interface
-
-Angular provides a CLI with commands for some common operations, like `ng build`, `ng serve`, `ng lint`, and `ng test`. (There's also `ng generate`, which lets you quickly make a new component or service without having to write out the boilerplate every time.)
+You should be able to access the Firebase Emulator Suite page at http://localhost:4000/
 
-In order to use the Angular CLI, you'll need to install the `@angular/cli` package globally:
+### Development Server
 
-```sh
-npm install -g @angular/cli
+To run the development server:
+```
+ng serve
 ```
 
-(If you're on a Unix system, you may need `sudo`.)
+Navigate to http://localhost:4200/. The app will automatically reload if you change any of the source files.
 
-## Setting up Firebase
 
-[Firebase][] is a set of cloud services provided by Google. We use a few of these services in Buzz About. Notably, we're using [Cloud Firestore][], a database that syncs between multiple clients in real time.
+## Testing
 
-When working on Buzz About, you'll want to make sure your school Google account gets added to our Firebase project. (Professor KK should be able to add you.)
+### Running unit tests
 
-### Installing the command-line interface
+To run the unit tests:
+```
+ng test
+```
 
-Like Angular, Firebase provides a set of command-line tools. You can get these by running
+### Running Cypress
 
-```sh
-npm install -g firebase-tools
+To build the app and run the Cypress end-to-end tests, start the Firebase emulators and run:
+```
+npm run e2e
 ```
 
-The first command that you want to run is probably `firebase login`. This will open a web browser prompting you to log in with your Google account.
+To open Cypress (so you can run the tests interactively), start both the Firebase emulators and the development server and run:
+```
+npm run cy:open
+```
 
-![PowerShell after running "firebase login". The program asks if it's okay to collect usage and error reporting, and then waits for authentication from the web browser.][i:Running firebase login]
+### Running Firebase tests
 
-![The browser window that "firebase login" spawns. It says that the Firebase CLI wants to view and manage your data across Google Cloud Platform services, view and administer all your Firebase data and settings, and view your cloud platform projects. At the bottom, there's an "Allow" button and a "Cancel" button.][i:Authenticating the Firebase CLI in the browser]
+With the Firebase emulators already running, run:
+```
+npm run firestore-specs
+```
 
-### Running the emulator
+To start the Firebase emulators and run through the Firebase tests:
+```
+npm run test-firestore
+```
 
```

Comment:
Hm... I'm wondering, do you think it would work to only mention `npm run test-firestore` here? (The `firestore-specs` command is sort of redundant with it.)

If not, is there a way to phrase this that more emphasizes the difference between `npm run firestore-specs` and `npm run test-firestore`?

Reply:
The `firestore-specs` command is useful for more quickly developing re-running specs by keeping the emulator running separately instead of waiting to start it everytime.

Reply:
Oh, right! Fair fair.